core/src/main/java/jenkins/install/InstallState.java

```diff
@@ -56,6 +57,7 @@
         new InstallState("UNKNOWN", false) {},
         new InstallState("INITIAL_SETUP_COMPLETED", false) {},
         new InstallState("CREATE_ADMIN_USER", false) {},
+        new InstallState("CONFIGURE_INSTANCE", false) {},
```

Comment:
I think that adding to this list will break deserialization from old Jenkins versions. The other items are here in the order in which they occurred in the source when they were anonymous classes, so inserting into the list will break the migration by causing any instances whose serialized state was `jenkins.install.InstallState$4`, which corresponds to `INITIAL_SECURITY_SETUP`, to be converted into the `CONFIGURE_INSTANCE` state when deserialized, which is probably not intentional.

Since no old instances will have this as the serialized state, I think you are fine with just the named class below and this line should be removed. Maybe also good to add something like "DO NOT ADD TO THIS LIST" to its Javadoc.